daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -90,53 +92,53 @@ private[io] class ByteArrayOrFileOutputStream(
             throw new FileIOException(
               "Unable to create temporary file in %s: %s".format(
                 tempDirPath.getPath,
-                e.getMessage(),
+                e.getMessage,
               ),
             )
         }
       val newStream: java.io.OutputStream = new java.io.FileOutputStream(maybeFile.get)
-      stream.flush
+      stream.flush()
```

Comment:
What is the best practice for inclusion of parentheses in parameterless function calls? It seems like we don't include them for most uses (see line 95 above), but is this a special case?

Reply:
The inclusion of empty parens in function definitions/calls is a semantic signal to all programmers that the function is not a pure function, that is, it performs a side effect or changes mutable state.  If a Scala function is declared/implemented with empty parens, the IDE will warn if calls are made without empty parens.  Likewise, if the function is defined as a parameterless function (without empty parens), the IDE will warn if function calls are made with empty parens.  However, Java doesn't implement these semantics itself, so Scala 2 is very liberal when it comes to mixing parameterless and empty-paren methods and calls.  You can add or remove empty parens in any place and the code will compile/run anyway although other programmers may misunderstand the function's semantics.  I think the Scala 3 compiler has become stricter and will check/enforce these semantics to a greater extent.

Reply:
One other wrinkle here is IDEs. My IntelliJ IDEA recommends scala style of including the () on a call if the method returns Unit (or void for Java), and not including the () on the call if the method returns a value. 

So toString is recommended to call without the parens because it returns a value. 
Ditto for getters of any kind. 

Of course for methods that both have side-effects, AND return a value, and have no args. In that case IntelliJ IDEA will recommend removing them, despite the method having side effects (because it really doesn't know about the side-effects, and I disagree.